onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.cpp

```diff
@@ -37,96 +46,181 @@ namespace Dml
+
+                    std::string partitionKernelPrefix = std::to_string(m_providerImpl->GetPartitionKernelPrefixVal()) + "_";
+                    m_providerImpl->IncreasePartitionKernelPrefixVal();
 
-                            continue;
+                    auto indexedSubGraph = DmlGraphFusionHelper::CreateIndexedSubGraph(partition.get(), partitionIndex, partitionKernelPrefix);
+
+                    // Create a map of which inputs are uploaded by the DML EP
+                    const uint32_t fusedNodeInputCount = gsl::narrow_cast<uint32_t>(indexedSubGraph.GetMetaDef()->inputs.size());
+                    std::vector<uint8_t> isInputsUploadedByDmlEP(fusedNodeInputCount);
+                    for (uint32_t index = 0; index < fusedNodeInputCount; ++index)
+                    {
+                        auto iter = isInitializerTransferable.find(indexedSubGraph.GetMetaDef()->inputs[index]);
+                        isInputsUploadedByDmlEP[index] = iter != isInitializerTransferable.end() ? true : false;
+                    }
+
+                    auto partitionNodePropsMap = DmlGraphFusionHelper::CreatePartitionNodePropsMap(
+                        graph,
+                        indexedSubGraph,
+                        std::move(graphNodePropertyMap));
+
+                    // Convert partitionONNXGraph into DML EP GraphDesc
+                    ComPtr<IDMLDevice> device;
+                    ORT_THROW_IF_FAILED(m_providerImpl->GetDmlDevice(device.GetAddressOf()));
+                    GraphDescBuilder::GraphDesc graphDesc = GraphDescBuilder::BuildGraphDesc(
+                        isInputsUploadedByDmlEP.data(),
+                        isInputsUploadedByDmlEP.size(),
+                        isInitializerTransferable,
+                        graph,
+                        indexedSubGraph,
+                        partitionNodePropsMap,
+                        device.Get(),
+                        m_providerImpl);
+
+                    // Compile the operator
+                    auto compiledPartition = DmlGraphFusionHelper::TryCreateCompiledOperator(
+                        graphDesc,
+                        indexedSubGraph,
+                        m_providerImpl);
+
+                    if (!compiledPartition)
+                    {
+                        if (indexedSubGraph.nodes.size() < 2)
+                        {
+                            // Fail early if even a single operator is too big to compile. This is highly unlikely.
+                            ORT_THROW_HR(E_INVALIDARG);
                         }
-                        isInitializerTransferable[input] = {tensor, true};
+
+                        // Insert a null operator next to the current one because it will get split in half during the next attempt
+                        compiledPartitionInfos.insert(compiledPartitionInfos.begin() + partitionIndex + partitionIndexOffset, {});
+                        ++partitionIndexOffset;
```

Comment:
Is compiledPartitionInfos supposed to be passed to BuildParition? Otherwise how compiledPartitionInfos will get used in next iteration?

Reply:
I changed the logic from when I was caching the successfully compiled partitions and forgot to remove those lines. It should make more sense now.